File: HandingOutTests/Codepal/Java/HandingOutTests.java
import java.util.*;

public class HandingOutTests {
    public static void main(String[] args) 
    {
        Scanner scanner = new Scanner(System.in);

        // Read the number of students
        int n = scanner.nextInt();

        // Create a 2D array to store the distances between students
        int[][] distances = new int[n][n];

        // Read the distances between each pair of students
        for (int i = 0; i < n * (n - 1) / 2; i++) {
            int student1 = scanner.nextInt() - 1;
            int student2 = scanner.nextInt() - 1;
            int distance = scanner.nextInt();
            distances[student1][student2] = distance;
            distances[student2][student1] = distance;
        }

        // Initialize the visited array and the starting node
        boolean[] visited = new boolean[n];
        visited[0] = true;
        int currentNode = 0;

        // Initialize the path and the total distance
        List<Integer> path = new ArrayList<>();
        path.add(currentNode + 1);
        int totalDistance = 0;

        // Repeat until all nodes have been visited
        while (path.size() < n) {
            // Find the nearest unvisited node
            int nearestNode = -1;
            int nearestDistance = Integer.MAX_VALUE;
            for (int i = 0; i < n; i++) {
                if (!visited[i] && distances[currentNode][i] < nearestDistance) {
                    nearestNode = i;
                    nearestDistance = distances[currentNode][i];
                }
            }

            // Mark the nearest node as visited and update the current node and the total distance
            visited[nearestNode] = true;
            currentNode = nearestNode;
            path.add(currentNode + 1);
            totalDistance += nearestDistance;
        }

        // Add the starting node to the end of the path and update the total distance
        path.add(1);
        totalDistance += distances[currentNode][0];

        // Print the total distance and the path
        System.out.println(totalDistance);
        for (int i = 0; i < path.size(); i++) {
            System.out.print(path.get(i));
            if (i < path.size() - 1) {
                System.out.print(" ");
            }
        }
        scanner.close();
    }
}

File: HandingOutTests/Codepal/C/HandingOutTests.c
#include <stdio.h>
#include <stdlib.h>
#include <limits.h>

#define MAX_STUDENTS 1000

int distance[MAX_STUDENTS][MAX_STUDENTS];
int visited[MAX_STUDENTS];
int path[MAX_STUDENTS];
int n_students;

int nearest_neighbor(int current_student) {
    int nearest_student = -1;
    int nearest_distance = INT_MAX;
    for (int i = 0; i < n_students; i++) {
        if (!visited[i] && distance[current_student][i] < nearest_distance) {
            nearest_student = i;
            nearest_distance = distance[current_student][i];
        }
    }
    return nearest_student;
}

void tsp() {
    int current_student = 0;
    visited[current_student] = 1;
    path[0] = current_student;
    int total_distance = 0;
    for (int i = 1; i < n_students; i++) {
        int next_student = nearest_neighbor(current_student);
        visited[next_student] = 1;
        path[i] = next_student;
        total_distance += distance[current_student][next_student];
        current_student = next_student;
    }
    total_distance += distance[current_student][0];
    printf("%d\n", total_distance);
    for (int i = 0; i < n_students; i++) {
        printf("%d ", path[i] + 1);
    }
    printf("1\n");
}

int main() {
    scanf("%d", &n_students);
    for (int i = 0; i < n_students; i++) {
        for (int j = 0; j < n_students; j++) {
            distance[i][j] = INT_MAX;
        }
    }
    int student1, student2, dist;
    for(int i = 1; i < n_students; i++) {
        for(int j = i; j < n_students; j++) {
            scanf("%d %d %d", &student1, &student2, &dist);
            distance[student1 - 1][student2 - 1] = dist;
            distance[student2 - 1][student1 - 1] = dist;
        }
    }
    tsp();
    return 0;
}

